x-pack/legacy/plugins/uptime/public/state/api/overview_filters.ts

```diff
@@ -4,47 +4,37 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ThrowReporter } from 'io-ts/lib/ThrowReporter';
+import { PathReporter } from 'io-ts/lib/PathReporter';
 import { isRight } from 'fp-ts/lib/Either';
 import { GetOverviewFiltersPayload } from '../actions/overview_filters';
-import { getApiPath, parameterizeValues } from '../../lib/helper';
 import { OverviewFiltersType } from '../../../common/runtime_types';
-
-type ApiRequest = GetOverviewFiltersPayload & {
-  basePath: string;
-};
+import { apiService } from './utils';
+import { API_URLS } from '../../../common/constants/rest_api';
 
 export const fetchOverviewFilters = async ({
-  basePath,
   dateRangeStart,
   dateRangeEnd,
   search,
   schemes,
   locations,
   ports,
   tags,
-}: ApiRequest) => {
-  const url = getApiPath(`/api/uptime/filters`, basePath);
-
-  const params = new URLSearchParams({
+}: GetOverviewFiltersPayload) => {
+  const queryParams = {
     dateRangeStart,
     dateRangeEnd,
-  });
+    schemes,
+    locations,
+    ports,
+    tags,
+    search,
+  };
 
-  if (search) {
-    params.append('search', search);
-  }
-
-  parameterizeValues(params, { schemes, locations, ports, tags });
+  const responseData = await apiService.get(API_URLS.FILTERS, queryParams);
 
-  const response = await fetch(`${url}?${params.toString()}`);
-  if (!response.ok) {
-    throw new Error(response.statusText);
-  }
-  const responseData = await response.json();
   const decoded = OverviewFiltersType.decode(responseData);
 
-  ThrowReporter.report(decoded);
+  PathReporter.report(decoded);
```

Comment:
Does this do anything? IIRC Throw reporter issues a throw, while `PathReporter` returns a value like what's shown here: https://github.com/gcanti/io-ts#error-reporters . I don't think we should be making this change, which is unrelated, in this PR.

Reply:
ThrowReporter is deprecated.

Reply:
Good to know. That said, I think we'll need `throw new Error(PathReporter.report(decoded))` the code as-is just generates a string and throws it away.

Reply:
we don't need to throw it, because we are not catching it. We are only checking for errors now.